app/components/candidate_interface/application_dashboard_component.html.erb

```diff
@@ -1,14 +1,11 @@
-<% if @application_form.ended_without_success? %>
-  <%= render CandidateInterface::CarryOverBannerComponent.new(application_form: @application_form) %>
-<% end %>
-
 <h1 class="govuk-heading-xl"><%= title %></h1>
 
 <div class="govuk-grid-row">
   <div class="govuk-grid-column-two-thirds">
     <%= render(CandidateInterface::ApplicationDashboardGuidanceComponent.new(application_form: @application_form)) %>
     <% if @application_form.ended_without_success? %>
       <%= render CandidateInterface::ApplyAgainCallToActionComponent.new(application_form: @application_form) %>
+      <%= render CandidateInterface::CarryOverCallToActionComponent.new(application_form: @application_form) %>
```

Comment:
Does this code mean tha potentially you could see 2 sets of inset text, one asking if you want to apply again, one asking if you want to carry over?

Reply:
@paulrobertlloyd You shouldn't ever see both. Each component has a `render?` method that should stop that happening (`CarryOverCallToActionComponent` only renders if the current recruitment cycle is different from the current application, `ApplyAgainCallToActionComponent` only renders when they are the same).

Reply:
Ah, I see. Yes, maybe a bit weird, but having the logic in the component is probably safer, if less clear in the view.